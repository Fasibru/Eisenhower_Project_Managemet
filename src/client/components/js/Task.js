import React from 'react';
import PropTypes from 'prop-types';

function Task(props) {
  const { className, title, description } = props;
  return (
    <div className={className} onDoubleClick={() => console.log('TEST')}>
      <p>{title}</p>
      <p>{description}</p>
    </div>
  );
}

Task.propTypes = {
  className: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
};

export default Task;
