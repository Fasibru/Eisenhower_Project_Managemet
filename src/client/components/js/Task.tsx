// import PropTypes from 'prop-types';
import * as React from 'react';
import { connect } from 'react-redux';
import { TaskType } from '../../../types/storeTypes';
import { TaskActionsTypes } from '../../../types/taskActionTypes';
import {
  openEditTaskPopup,
  populateEditTaskForm,
} from '../../actions/actionsTasks';

interface TaskProps {
  task: TaskType;
  className: string;
  openEditTaskPopup(): TaskActionsTypes;
  populateEditTaskForm(task: TaskType): TaskActionsTypes;
}

// tslint:disable-next-line: variable-name
export const Task: React.FC<TaskProps> = ({
  task,
  className,
  // tslint:disable: no-shadowed-variable
  openEditTaskPopup,
  populateEditTaskForm,
  // tslint:enable: no-shadowed-variable
}) => {
  const populateEditTask = () => {
    openEditTaskPopup();
    populateEditTaskForm(task);
  };

  return (
    <div
      className={className}
      onDoubleClick={populateEditTask}
    >
      <p className="task__title--bold">{task.title}</p>
      <p style={{ whiteSpace: 'pre-wrap' }}>{task.description}</p>
    </div>
  );
};

// Task.propTypes = {
//   className: PropTypes.string.isRequired,
//   task: PropTypes.shape({
//     title: PropTypes.string.isRequired,
//     description: PropTypes.string.isRequired,
//     category: PropTypes.string.isRequired,
//     rank: PropTypes.number.isRequired,
//     _id: PropTypes.string.isRequired,
//   }).isRequired,
//   populateEditTaskForm: PropTypes.func.isRequired,
//   openEditTaskPopup: PropTypes.func.isRequired,
// };

export default connect(null, { openEditTaskPopup, populateEditTaskForm })(Task);
