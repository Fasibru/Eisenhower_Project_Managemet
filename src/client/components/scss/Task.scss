@import './variables';

.task {
  margin: 4px 8px 4px 8px;
  padding: 8px;
  overflow-wrap: break-word;
  word-wrap: break-word;
  hyphens: auto;
  background-color: $component-color-Task;
}

.task:first-child {
  margin-top: 8px;
}

.task:last-child {
  margin-bottom: 8px;
}

.task:hover {
  background: $hover-color;
}

.task--red {
  border: 1px solid red;
}

.task--orange {
  border: 1px solid orange;
}

.task--green {
  border: 1px solid green;
}

.task--lightgray {
  border: 1px solid lightgray;
}

.task__title--bold {
  font-weight: bold;
}

.task__interaction-icons {
  display: flex;
  margin-bottom: 0;
  margin-top: 0;
  color: $color-dark-font;
  font-size: 1rem;
}

.task__interaction-icon--push-right{
  margin-left: auto;  
}

.task__interaction-icon--increased-margin{
  margin: 0 15px;
}

.task__interaction-icon-button {
  border: 0;
  padding: 0;
  background-color: transparent;
  color: $color-dark-font;
  font-size: 1rem;
}

.task__interaction-icon-button:hover {
  color: #000000;
  cursor: pointer;
}

// Medium screens breakpoint
@media only screen and (min-width: 992px) {
  .task--orange, .task--lightgray {
    margin-left: 4px;
  }

  .task--red, .task--green {
    margin-right: 4px;
  }
}

// Large screens breakpoint
@media only screen and (min-width: 1400px) {
  .task--orange, .task--green {
    margin-left: 4px;
    margin-right: 4px;
  }
}