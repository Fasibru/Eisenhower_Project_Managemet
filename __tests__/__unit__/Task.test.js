import React from 'react';
import { shallow } from 'enzyme';
import Task from '../../src/client/components/js/Task';

describe('Testing <Task />', () => {
  const fnDoubleClick = jest.fn();
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Task
      className="test"
      key={1}
      task={{
        _id: '2',
        rank: -999,
        category: 'A',
        title: 'Task Title',
        description: 'Task Description',
        completed: false,
      }}
      populateEditTask={fnDoubleClick}
    />);
  });

  it('Renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('Should populate "title" and "description" based on provided props', () => {
    const pTitle = wrapper.find('p').at(0);
    const pDescription = wrapper.find('p').at(1);

    expect(pTitle.text()).toEqual('Task Title');
    expect(pDescription.text()).toEqual('Task Description');
  });

  it('Should run "populateEditTask" on double click', () => {
    const div = wrapper.find('div');
    div.simulate('doubleClick');

    expect(fnDoubleClick).toHaveBeenCalledTimes(1);
  });
});
