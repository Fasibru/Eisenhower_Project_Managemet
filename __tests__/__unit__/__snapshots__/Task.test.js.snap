// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <Task /> Renders correctly 1`] = `
<div
  className="test"
  onDoubleClick={[Function]}
>
  <p>
    Task Title
  </p>
  <p>
    Task Description
  </p>
</div>
`;
